### PR TITLE
feat: use percentage discount instead of absolute

### PR DESCRIPTION
**Pull request by ChristophHi**

### CODE REVIEW

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 1]
```diff
@@ -49,7 +48,7 @@ export class QuoteSummarySellerEditComponentService {
   }
 
   /**
-   * Retrieves formatter according to current locale and currency
+   * Retrieves formatter according to current locale and unit
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts]: Could you, please, add a dot at the end of the comment sentence and an empty line between the comment and @param / @returns? Thanks!

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 2]
```diff
@@ -59,25 +58,20 @@ export class QuoteSummarySellerEditComponentService {
   }
 
   /**
-   * Retrieves localization elements according to current locale and currency
+   * Retrieves localization elements according to current locale and unit
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts]: The same here, please.

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 3]
```diff
@@ -149,32 +133,15 @@ export class QuoteSummarySellerEditComponentService {
...
-    const numberOfDecimalPlaces = 2;
-    const maximum = Math.max(
-      quote.totalPrice.value ?? 1,
-      quote.quoteDiscounts?.value ?? 1
-    );
-    return Math.floor(Math.log10(maximum)) + 1 + numberOfDecimalPlaces;
-  }
-
-  protected checkAndReportCurrencyIfMissing(
+  protected checkAndReportPercentageSignIfMissing(
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts]: What about `checkAndReportMissedPercentageSign`?
=> the method checks if the percentage sign is missed and
=> reports it accordingly

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 4]
```diff
@@ -149,32 +133,15 @@ export class QuoteSummarySellerEditComponentService {
...
-    currency?: string
+    percentageSign?: string
   ): LocalizationElements {
-    if (currency) {
-      return { locale, formatter, currencySymbol: currency };
+    if (percentageSign) {
+      return { locale, formatter, percentageSign: percentageSign };
     } else {
-      throw new Error('Currency must have symbol or ISO code');
+      throw new Error('No percentage sign present');
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts:150]: Do we want to throw a console error or show a global message as well?

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 5]
```diff
@@ -245,13 +210,26 @@ export class QuoteSummarySellerEditComponentService {
     return isError ? { wrongFormat: {} } : null;
   }
 
-  protected performValidationAccordingToMetaData(
+  /**
+   * Returns maximum number of decimal places for the percentage discount.
+   * The value is read from the configuration, refer to attribute quote/maximumDecimalsForPercentageDiscount.
+   * In case this configuration is not present, default value is 8
+   *
+   * @returns Maximum number of decimal places for percentage discount
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts:224]: Please, add @ protected annotation.

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 6]
```diff
@@ -245,13 +210,26 @@ export class QuoteSummarySellerEditComponentService {
...
+
+  protected performValidationForPercentageValue(
     input: string,
     groupingSeparator: string,
-    decimalSeparator: string,
-    numberTotalPlaces: number
+    decimalSeparator: string
   ): boolean {
-    const numberDecimalPlaces = 2;
+    const numberDecimalPlaces = this.retrieveMaxNumberOfDecimalPlaces();
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts]: What about `numberOfDecimalPlaces`?

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts [hunk 7]
```diff
@@ -245,13 +210,26 @@ export class QuoteSummarySellerEditComponentService {
...
     const regexEscape = '\\';
     const search: RegExp = new RegExp(regexEscape + groupingSeparator, 'g');
     const woGrouping = input.replace(search, '');
```
Comment (Larisa-Staroverova) [inline feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts]: Could you explain to me what `woGrouping` means?

**Review by Larisa-Staroverova — CHANGES_REQUESTED** [review]
@ChristophHi, could you, please, check all comments and add a dot at the end of the comment sentence and an empty line between the comment and @param / @returns? Thanks!

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts (hunk 4 above)
Comment (ChristophHi) [reply feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts:150]: I guess that's not needed, as such an error would be reflected in the console. Moreover that should not happen at runtime for this formatter (we request a percent formatter)

File: feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts (hunk 7 above)
Comment (ChristophHi) [reply feature-libs/quote/components/summary/seller-edit/quote-summary-seller-edit.component.service.ts]: stands for 'withoutGroupingSeparator', will rename

**Review by Larisa-Staroverova — APPROVED** [review]

**Pull request merged**